DylibBinder/XmlGenerator.cs

```diff
@@ -0,0 +1,258 @@
+﻿using System;
+using System.Linq;
+using System.Xml;
+
+namespace DylibBinder {
+	public class XmlGenerator {
+		public XmlGenerator (DBTopLevel dBTopLevel, string outputPath)
+		{
+			XmlWriter writer = CreateWriter (outputPath);
+			WriteIntro (writer, dBTopLevel);
+			WriteTypeDeclarations (writer, dBTopLevel);
+			CloseWriter (writer);
+		}
+
+		XmlWriter CreateWriter (string outputPath)
+		{
+			XmlWriterSettings settings = new XmlWriterSettings ();
+			settings.Indent = true;
+			settings.IndentChars = "   ";
+			return XmlWriter.Create (outputPath, settings);
+		}
+
+		void WriteIntro (XmlWriter writer, DBTopLevel dBTopLevel) {
+			writer.WriteStartDocument ();
+			writer.WriteStartElement ("xamreflect");
+			writer.WriteAttributeString ("version", dBTopLevel.XmlVersion);
+
+			writer.WriteStartElement ("modulelist");
+
+			writer.WriteStartElement ("module");
+			writer.WriteAttributeString ("name", dBTopLevel.ModuleName);
+			writer.WriteAttributeString ("swiftVersion", dBTopLevel.SwiftVersion);
+		}
+
+
+		void WriteTypeDeclarations (XmlWriter writer, DBTopLevel dBTopLevel)
+		{
+			foreach (var typeDeclaration in dBTopLevel.DBTypeDeclarations.TypeDeclarations) {
+
+				writer.WriteStartElement ("typedeclaration");
+				WriteMultipleAttributeStrings (writer, ("kind", typeDeclaration.Kind),
+											   ("name", typeDeclaration.Name), ("accessibility", typeDeclaration.Accessibility),
+											   ("isDeprecated", typeDeclaration.IsDeprecated), ("isUnavailable", typeDeclaration.IsUnavailable),
+											   ("isObjC", typeDeclaration.IsObjC), ("isFinal", typeDeclaration.IsFinal));
```

Comment:
lets indent a little less hehe

Reply:
Idk why it shows up like that here.. maybe its a weird setting on my vsfm? haha